### PR TITLE
#79 code style improvement

### LINKED ISSUE #79: Possible Improvement: Code Style
While writing tests for #76, I've noticed the quotes used in test/restClient.spec.js vary throughout the file.

For example, this 'describe' block uses different quoting for each of the two 'it' blocks:
```js
  describe('when called with CREATE', function () {
      //snip
    it("calls the client's create method with the data in params", function () {
      //snip
    });

    it('returns the data returned by the client in a "data" object', function () {
      //snip
    });
  });
```

I have a couple of questions raised by this:

- Is there a designated style guide this project should adhere to? Personally I've always liked the [Airbnb Javascript Style Guide](https://github.com/airbnb/javascript) and would suggest this
- Is it worth considering adding linting somewhere? Possibly in the test config so it's picked up by CI checks before merging PRs?

For the purposes of getting #76 done now, I'm working under the assumption that single quotes are the preferred option.

### PR DESCRIPTION
**Pull request by sgobotta**
This PR provides:

+ ESLint configuration to enforce coding style consistency
+ closes #79

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -58,6 +58,11 @@
     "babel-preset-stage-0": "^6.16.0",
     "babel-register": "^6.26.0",
     "chai": "^3.5.0",
+    "eslint": "^5.7.0",
+    "eslint-config-airbnb": "^17.1.0",
+    "eslint-plugin-import": "^2.14.0",
+    "eslint-plugin-jsx-a11y": "^6.1.2",
+    "eslint-plugin-react": "^7.11.1",
```
Comment (josx) [inline package.json]: Do we need this dep?

File: package.json [hunk 2]
```diff
@@ -58,6 +58,11 @@
     "babel-preset-stage-0": "^6.16.0",
     "babel-register": "^6.26.0",
     "chai": "^3.5.0",
+    "eslint": "^5.7.0",
+    "eslint-config-airbnb": "^17.1.0",
+    "eslint-plugin-import": "^2.14.0",
+    "eslint-plugin-jsx-a11y": "^6.1.2",
```
Comment (josx) [inline package.json]: Do we need this dep?

**Comment by josx** [comment]
In  #79 We mention lint-staged, can you add it? 

In package.json something like this:
```
"devDependencies": {
"lint-staged": "^7.2.2",
},
"gitHooks": {
    "pre-commit": "lint-staged"
  },
  "lint-staged": {
    "*.js": [
      "vue-cli-service lint",
      "git add"
    ]
  }
```

**Comment by sgobotta** [comment]
I will get every eslint issue fixed and then implement the lint-staged feature

**Comment by josx** [comment]
ok please go on! :+1:

**Comment by sgobotta** [comment]
Test won't pass since yarn is not installing eslint correctly. Following [this](https://github.com/airbnb/javascript/issues/1508) to fix yarn install.

**Pull request merged**